core/src/main/java/org/frankframework/http/rest/ApiListenerServlet.java

```diff
@@ -600,16 +602,20 @@ else if(segment.startsWith("{") && segment.endsWith("}")) {
 					response.setStatus(statusCode);
 				}
 
-				/*
-				 * Finalize the pipeline and write the result to the response
-				 */
-				final boolean outputWritten = writeToResponseStream(response, result);
-				if (!outputWritten) {
-					LOG.debug("No output written, set content-type header to null");
-					response.resetBuffer();
-					response.setContentType(null);
+				if (method != ApiListener.HttpMethod.HEAD) {
+					/*
+					 * Finalize the pipeline and write the result to the response
+					 */
+					final boolean outputWritten = writeToResponseStream(response, result);
+					if (!outputWritten) {
+						LOG.debug("No output written, set content-type header to null");
+						response.resetBuffer();
+						response.setContentType(null);
+					}
+				} else {
+					response.setContentLength(0);
+					response.addHeader("content-length", "0");
```

Comment:
Now you always set content-length to 0 when method = HEAD?
I thought that that goes against the intention?